modules/misc/opengl.nix

```diff
@@ -0,0 +1,62 @@
+{ config, pkgs, lib, ... }:
+
+let
+  nixGL = (import (builtins.fetchTarball
+    "https://github.com/nix-community/nixGL/archive/main.tar.gz") {
+      inherit pkgs;
+    });
```

Comment:
This is used in the default option below, but I'm unsure how to create an option that depends on an external package and only pull the external package when a user needs it. In its current state, it always pulls it in which means `home-manager` becomes dependent on nixgl, even for users who use home-manager in a headless environment.